### ISSUE TITLE
build role conflict resolution into axe.commons.aria.getRole

### ISSUE THREAD

**Issue by straker** [dequelabs/axe-core#2212]
This should replace the exception we build into axe.commons.text.titleText.

https://www.w3.org/TR/wai-aria-1.1/#conflict_resolution_presentation_none

**Comment by straker** [comment]
See also https://github.com/w3c/aria/issues/1270

**Comment by padmavemulapati** [comment]
when role=presentation and aria-attributes are properly defined no failures seeing: 
for  ex: `<h1 role="presentation" aria-haspopup="true"> Sample Content </h1> ` is passing no failure found
but ` <h1 role="presentation" aria-level="2"> Sample  Content </h1>` is failing for `aria-allowed-attr` - ruleid

**Comment by jeankaplansky** [comment]
no user-facing docs required.
